### ISSUE TITLE
Bar autohide + Super key

### ISSUE THREAD

**Issue by lucasgabmoreno** [linuxmint/cinnamon#11035]
```
 * Cinnamon 5.4.9
 * Distribution: Linux Mint Debian Edition LMDE 5
 * Intel HD 5500(i915) + Nvidia 940M (nvidia-driver)
 * 64 bit
 ```

**Issue**

After updating to last Cinnamon version, when press Super (Winkey) in autohide bar, this overlaps Menu.

![2022-08-10_19-04](https://user-images.githubusercontent.com/83520011/184028506-2233432f-1cc3-4100-a1ab-140e7bfcf043.png)

**Steps to reproduce**
Super (Winkey)

**Expected behaviour**
No overlap Menu and bar

**Comment by mtwebster** [comment]
Hi, can you try out the Mint cinnamon themes (any of them) to see if this happens there?  This may be a theme issue.

**Comment by lucasgabmoreno** [comment]
Nope, same problem with Mint-Y-Dark (in different computers).

![Captura de pantalla de 2022-08-10 21-04-56](https://user-images.githubusercontent.com/83520011/184043507-d5da0ca7-39e0-453a-a531-0b4ad079569b.png)

**Comment by rcalixte** [comment]
I've noticed this as well but haven't had a chance to dig into this part of the code yet. Essentially what is happening is when the panel is set to auto-hide and the Super key is pressed (or whatever Open Menu accelerators are configured) the Menu is displayed before the panel is unhidden which causes the panel to appear above the bottom of the menu dialog box. The same thing can happen if the Menu has "Open the menu when I move my mouse over it" enabled.

It seems like a race condition where the panel needs to be expanded first and only once fully expanded, then the Menu should be opened.

@lucasgabmoreno A workaround I've found when needed is to mouse over the panel to expand it before opening the Menu or disabling the option "Force the panel to be visible when opening the menu" in the settings.

**Comment by rcalixte** [comment]
Duplicate of #10970

**Comment by lucasgabmoreno** [comment]
Thanks @rcalixte! But I can't find "Force the panel to be visible when opening the menu" en settings.
![Captura de pantalla de 2022-08-13 10-31-36](https://user-images.githubusercontent.com/83520011/184496367-ad672262-99ed-40fc-a4aa-abe6cca5f5ce.png)

**Comment by mtwebster** [comment]
Right-click the start menu button and select 'Configure' - it's in the applet's settings, not panels.

**Comment by mtwebster** [comment]
In the past, it was the same behavior as current, except the menu shifted once the panel was fully open. I had a look at this yesterday - 

- Waiting to show the menu until the panel reveals itself would take too long.
- Eliminating the panel animation when the menu opens is really not viable currently without a lot of rework to the panel or a lot of ugly hacks (there are delays and callbacks involved that make 'show panel' -> 'panel shown' non-synchronous.

The simplest solution for now is to mimic the behavior in previous versions and have the menu shift when the panel opens.

**Comment by lucasgabmoreno** [comment]
Thanks! I found them but doesn't solve the problem.

**Comment by nhamawi** [comment]
This is still a problem today in the latest release of Cinnamon and even worse if you dock the taskbar panel to the top of the screen, in which case, the search box is partially obscured.

![Screenshot from 2023-03-16 12-25-56](https://user-images.githubusercontent.com/2650762/225692564-48c50d76-6ff0-45f2-861b-303e8cf47a98.png)

**Comment by fredcw** [comment]
This issue was fixed by 11788 above and can be closed.
